pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -0,0 +1,174 @@
+package nodenetworkconfigurationpolicy
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var (
+	log = logf.Log.WithName("controller_nodenetworkconfigurationpolicy")
+)
+
+// Add creates a new NodeNetworkConfigurationPolicy Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileNodeNetworkConfigurationPolicy{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+func getNodeName() (string, error) {
+	nodeName := os.Getenv("NODE_NAME")
+	if len(nodeName) == 0 {
+		return nodeName, fmt.Errorf("no NODE_NAME environment variable")
+	}
+	return nodeName, nil
+}
+
+func matches(nodeSelector map[string]string, labels map[string]string) bool {
+	for key, value := range nodeSelector {
+		if foundValue, hasKey := labels[key]; !hasKey || foundValue != value {
+			return false
+		}
+	}
+	return true
+}
+
+func nodeSelectorMatchesThisNode(cl client.Client, eventObject runtime.Object) bool {
+	nodeName, err := getNodeName()
+	if err != nil {
+		log.Info("NODE_NAME not found for pod")
+		return false
+	}
+	node := corev1.Node{}
+	err = cl.Get(context.TODO(), types.NamespacedName{Name: nodeName}, &node)
+	if err != nil {
+		log.Info("Cannot find corev1.Node", "nodeName", nodeName)
+		return false
+	}
+
+	policyNodeSelector := eventObject.(*nmstatev1alpha1.NodeNetworkConfigurationPolicy).Spec.NodeSelector
```

Comment:
node selector is not mandatory, is it? This could fail (or the following, passing nil as selector)

Reply:
it's not a pointer they cannot pass nil there, if they don't configure it it will be an empty map I think

Reply:
Ok, I will add a check there, the idea is that if they don't specify nodeSelector all nodes will be selected, but let's put that as a unit test.

Reply:
Looks like map is never nil
https://play.golang.org/p/gPZB49dQUzF